packages/compass-crud/src/components/editable-document.tsx

```diff
@@ -200,7 +221,7 @@ class EditableDocument extends React.Component<
     if (!this.state.editing && !this.state.deleting) {
       return (
         <DocumentList.DocumentActionsGroup
-          onEdit={this.handleEdit.bind(this)}
+          onEdit={this.handleStartEditing.bind(this)}
```

Comment:
> Looks like there are two start editing functions here, is that intended? Looks like it used to set state and now calls it on the document.

Its now intended to call startEditing on HadronDocument because we shifted the document editing state over to HadronDocument itself. On call of that method, we update the state in the HadronDocument and also emit an event which gets picked up in this component and that updates the state of the component itself. This is done so we can preserve the state of editing / deleting when the list unmounts out of view components on scroll.

> Something we probably want to look at is when a user starts editing a document, then changes the documents shown per page. The document is still flagged for edits, but it gets back the non-edit actions:

This might be an issue, I might have missed writing a test for this. Will check it and cover that as well.

Reply:
> This might be an issue, I might have missed writing a test for this. Will check it and cover that as well. Thank you for spotting <3

Ok so it turned out that component at the index retains its editing state but the underlying HadronDocument has obviously changed because of the refresh due to the change in page size. Fixed this in [fbb5ab7](https://github.com/mongodb-js/compass/pull/6069/commits/fbb5ab7238ac2e4dd4b5b261317ef2b5f22e36c1) and added a test for this as well.